File: backend/LimsAuth.Api/Models/TeachingTask.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LimsAuth.Api.Models;

/// <summary>
/// 教学任务实体
/// </summary>
[Table("teaching_tasks")]
public class TeachingTask
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    /// <summary>
    /// 学期ID
    /// </summary>
    [Required]
    [Column("semester_id")]
    public Guid SemesterId { get; set; }

    /// <summary>
    /// 课程ID
    /// </summary>
    [Required]
    [Column("course_id")]
    public Guid CourseId { get; set; }

    /// <summary>
    /// 班级ID
    /// </summary>
    [Required]
    [Column("class_id")]
    public Guid ClassId { get; set; }

    /// <summary>
    /// 任务类型(主讲/辅导/实验)
    /// </summary>
    [Column("task_type")]
    [MaxLength(20)]
    public string TaskType { get; set; } = "主讲";

    [Column("description")]
    [MaxLength(500)]
    public string? Description { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // 导航属性
    public virtual Semester Semester { get; set; } = null!;
    public virtual Course Course { get; set; } = null!;
    public virtual Class Class { get; set; } = null!;
    public virtual ICollection<TeachingTaskTeacher> Teachers { get; set; } = new List<TeachingTaskTeacher>();
}

/// <summary>
/// 教学任务教师关联
/// </summary>
[Table("teaching_task_teachers")]
public class TeachingTaskTeacher
{
    [Column("teaching_task_id")]
    public Guid TeachingTaskId { get; set; }

    [Column("teacher_id")]
    public Guid TeacherId { get; set; }

    /// <summary>
    /// 是否主讲教师
    /// </summary>
    [Column("is_main_teacher")]
    public bool IsMainTeacher { get; set; } = false;

    [Column("assigned_at")]
    public DateTime AssignedAt { get; set; } = DateTime.UtcNow;

    // 导航属性
    public virtual TeachingTask TeachingTask { get; set; } = null!;
    public virtual User Teacher { get; set; } = null!;
}

File: mobile/lib/models/teaching_task.dart
import 'semester.dart';
import 'course.dart';
import 'class_model.dart';
import 'user.dart';

class TeachingTask {
  final String id;
  final String? semesterId;
  final String? courseId;
  final String? classId;
  final String? taskType;
  final String? description;
  final bool isActive;
  final String? createdAt;
  final Semester? semester;
  final Course? course;
  final ClassModel? classModel;
  final List<TeachingTaskTeacher>? teachers;

  TeachingTask({
    required this.id,
    this.semesterId,
    this.courseId,
    this.classId,
    this.taskType,
    this.description,
    this.isActive = true,
    this.createdAt,
    this.semester,
    this.course,
    this.classModel,
    this.teachers,
  });

  factory TeachingTask.fromJson(Map<String, dynamic> json) {
    return TeachingTask(
      id: json['id']?.toString() ?? '',
      semesterId: json['semesterId']?.toString(),
      courseId: json['courseId']?.toString(),
      classId: json['classId']?.toString(),
      taskType: json['taskType']?.toString(),
      description: json['description']?.toString(),
      isActive: json['isActive'] as bool? ?? true,
      createdAt: json['createdAt']?.toString(),
      semester: json['semester'] != null
          ? Semester.fromJson(json['semester'] as Map<String, dynamic>)
          : null,
      course: json['course'] != null
          ? Course.fromJson(json['course'] as Map<String, dynamic>)
          : null,
      classModel: json['class'] != null
          ? ClassModel.fromJson(json['class'] as Map<String, dynamic>)
          : null,
      teachers: (json['teachers'] as List<dynamic>?)
          ?.map((e) => TeachingTaskTeacher.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }

  Map<String, dynamic> toJson() => {
        'semesterId': semesterId,
        'courseId': courseId,
        'classId': classId,
        'taskType': taskType,
        'description': description,
        'teacherIds': teachers?.map((t) => t.teacherId ?? '').where((id) => id.isNotEmpty).toList(),
      };
}

class TeachingTaskTeacher {
  final String? teachingTaskId;
  final String? teacherId;
  final bool isMainTeacher;
  final String? assignedAt;
  final User? teacher;

  TeachingTaskTeacher({
    this.teachingTaskId,
    this.teacherId,
    this.isMainTeacher = false,
    this.assignedAt,
    this.teacher,
  });

  factory TeachingTaskTeacher.fromJson(Map<String, dynamic> json) {
    return TeachingTaskTeacher(
      teachingTaskId: json['teachingTaskId']?.toString(),
      teacherId: json['teacherId']?.toString(),
      isMainTeacher: json['isMainTeacher'] as bool? ?? false,
      assignedAt: json['assignedAt']?.toString(),
      teacher: json['teacher'] != null
          ? User.fromJson(json['teacher'] as Map<String, dynamic>)
          : null,
    );
  }
}
